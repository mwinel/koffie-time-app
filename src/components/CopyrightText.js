import React from "react";
import { Link } from "react-router-dom";

const CopyrightText = () => {
  return (
    <div className="footer-bottom py-4">
      <div className="container">
        <div className="copyright-text">
          <small className="mr-3">Â© KoffieTime 2019</small>
          <small className="mr-3">
            <Link to="/" className="copyright-text-link">Terms</Link>
          </small>
          <small className="mr-3">
            <Link to="/" className="copyright-text-link">Privacy</Link>
          </small>
        </div>
      </div>
    </div>
  );
};

export default CopyrightText;
