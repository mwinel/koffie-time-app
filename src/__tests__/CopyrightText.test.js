import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";
import { BrowserRouter } from "react-router-dom";
import CopyrightText from "../components/CopyrightText";

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it("renders copyright-text component", () => {
  act(() => {
    render(
      <BrowserRouter>
        <CopyrightText />
      </BrowserRouter>,
      container
    );
  });
  expect(container.innerHTML).toBeDefined();
  expect(container.innerHTML).toContain("Â© KoffieTime 2019");
  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
