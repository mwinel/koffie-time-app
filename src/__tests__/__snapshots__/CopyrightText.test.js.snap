// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders copyright-text component 1`] = `
"<div class=\\"footer-bottom py-4\\">
  <div class=\\"container\\">
    <div class=\\"copyright-text\\"><small class=\\"mr-3\\">Â© KoffieTime 2019</small><small class=\\"mr-3\\"><a class=\\"copyright-text-link\\" href=\\"/\\">Terms</a></small><small class=\\"mr-3\\"><a class=\\"copyright-text-link\\" href=\\"/\\">Privacy</a></small></div>
  </div>
</div>"
`;
